src/app/pipes/meql.pipe.spec.ts

```diff
@@ -0,0 +1,356 @@
+    field: string,
+    operator: string,
+    value: any
+  ) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+  )
+   */
+    const query = {
+      condition: 'and',
+      rules: [{ field, operator, value }],
+    };
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = ')';
+    return [pipe.transform(query), `${line1}${line2}${line3}`];
+  }
+
+  public static doubleConditionTest(pipe: MeqlPipe, query: any) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+    and "Field2" = "Value2"
+  )
+   */
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = `${tab}${query.condition} "${query.rules[1].field}" ${query.rules[1].operator} "${query.rules[1].value}"${newline}`;
+    const line4 = ')';
+    return [pipe.transform(query), `${line1}${line2}${line3}${line4}`];
+  }
+
+  public static doubleRulesetTest(pipe: MeqlPipe, query: any) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+    and "Field2" = "Value2"
+    and (
+        "Field3" = "Value3"
+        or "Field4" = "Value4"
+    )
+  )
+   */
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = `${tab}${query.condition} "${query.rules[1].field}" ${query.rules[1].operator} "${query.rules[1].value}"${newline}`;
+    const line4 = `${tab}${query.condition} (${newline}`;
+    const line5 = `${tab}${tab}"${query.rules[2].rules[0].field}" ${query.rules[2].rules[0].operator} "${query.rules[2].rules[0].value}"${newline}`;
+    const line6 = `${tab}${tab}${query.rules[2].condition} "${query.rules[2].rules[1].field}" ${query.rules[2].rules[1].operator} "${query.rules[2].rules[1].value}"${newline}`;
+    const line7 = `${tab})${newline}`;
+    const line8 = ')';
+    return [
+      pipe.transform(query),
+      `${line1}${line2}${line3}${line4}${line5}${line6}${line7}${line8}`,
+    ];
+  }
+}
+
+describe('JqlPipe', () => {
+  let pipe: MeqlPipe;
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [MeqlPipe],
+    });
+    pipe = TestBed.inject(MeqlPipe);
+  });
+
+  it('create an instance', () => {
+    expect(pipe).toBeTruthy();
+  });
+
+  it('transforms field equal to string', () => {
+    const testResult = meqlTestHelper.singleConditionTest(
+      pipe,
+      'Simple String Field',
+      '=',
+      'Some Text'
+    );
+    expect(testResult[0]).toBe(testResult[1]);
+  });
+
+  it('transforms field not equal to string', () => {
+    const testResult = meqlTestHelper.singleConditionTest(
+      pipe,
+      'Simple String Field',
+      '!=',
+      'Some Text'
+    );
+    expect(testResult[0]).toBe(testResult[1]);
+  });
+
+  it('transforms field equal to number', () => {
```

Comment:
If you're familiar with data-driven tests, then there are ways in Jest to write those and parameterise some of your tests:

```ts
// Run this test on each value passed to each()
it.each([1, 2, 3])('should print %p'), (value) => {
  // Use the value here
});
```

I was wondering if some of the repetition I've seen could be applied here.